clash-lib/src/Clash/Core/TermInfo.hs

```diff
@@ -183,7 +183,8 @@ piResultTyMaybe m ty arg
   | Just ty' <- coreView1 m ty
   = piResultTyMaybe m ty' arg
   | FunTy a res <- tyView ty
-  = if debugIsOn && not (aeqType a arg) then error [I.i|
+  -- TODO Remove when coreview is not here: types should be aeq anyway
```

Comment:
I don't get what this comment is trying to convey. What needs to be removed?

Reply:
The thing, when the other thing isn't, _what's so hard to understand?_ /s